.NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs

```diff
@@ -232,11 +271,21 @@ private string PreprocessStrWithNumberExtracted(string str, out Dictionary<int,
                 }
                 else
                 {
-                    string originalText = str.Substring(start, end - start + 1);
-                    ret += replaceText;
-                    for (int i = 0; i < replaceText.Length; i++)
+                    if (type > 0)
+                    {
+                        ret += replaceText;

```

Comment:
This two conditions only differ by replaceText/replaceFracText.
It's best to assign the right value to a var and use in the code, no need to duplicate the whole thing.